src/ucs/sys/sock.h

```diff
@@ -469,6 +469,19 @@ int ucs_sockaddr_is_inaddr_any(struct sockaddr *addr);
 ucs_status_t ucs_sockaddr_copy(struct sockaddr *dst_addr,
                                const struct sockaddr *src_addr);
 
+/**
+ * Copy the src_addr sockaddr to dst_addr sockaddr. The length to copy is
+ * the size of the src_addr sockaddr. If src_addr contains an address family
+ * that is not IPv4 or IPv6, the dst_addr sockaddr structure will be set to
+ * all zeroes.
```

Comment:
name and description of this func looks inconsistent. Since this file is part of installable UCS/API, IMO such helper function should be defined where it's used.

Reply:
it's used from both uct and ucp.

Reply:
BTW, why it's important to zero non-IP addr? maybe better to pass it as is in sockaddr storage and allow to user handle it? smth like this:
```
void ucp_sockaddr_copy_storage(dst, src);
```

Reply:
if the address family isn't valid (ipv4 or ipv6) we don't know the size to copy.
setting to zero is a way for us to give an indication to the upper layer that something is 'wrong'.
iodemo, for example, already handles it by checking the address family before printing the sockaddr.
btw, it's the same as rdma_get_local/remote_addr behave in case they cannot return the sockaddr:
from rdma_get_local_addr:

`RETURN VALUE`
`Returns a pointer to the local sockaddr address of the rdma_cm_id.  If the rdma_cm_id is not bound to an address, the contents of the sockaddr structure will be set to all zeroes.
`

Reply:
this is why we should copy it as sockaddr storage

Reply:
not all transports have to set to zeros like rdmacm does.

Reply:
so we should handle it in `ep_query` see conversation below

Reply:
maybe return also the address length from both uct and ucp? same as getsockname()